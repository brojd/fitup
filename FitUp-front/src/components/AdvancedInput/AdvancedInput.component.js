import React, { PropTypes } from 'react';
import styles from './AdvancedInput.stylesheet.css';
import classNames from 'classnames';
import Select from 'react-select';
import compact from 'lodash/compact';
import uniqBy from 'lodash/uniqBy';

class AdvancedInput extends React.Component {
  constructor() {
    super();
    this._handleChange = this._handleChange.bind(this);
    this._handleFocus = this._handleFocus.bind(this);
    this._handleBlur = this._handleBlur.bind(this);
    this._prepareItems = this._prepareItems.bind(this);
    this.state = {
      textInputFocused: false,
      choices: []
    };
  }
  _handleChange(e) {
    this.setState({
      choices: e
    });
    this.props.onChange(e);
  }
  _handleFocus() {
    this.props.onFocus();
  }
  _handleBlur() {
    this.props.onBlur();
  }
  _prepareItems() {
    let items = this.props.items.map((item) => {
      let fullName;
      if (item.surname) {
        fullName = `${item.name} ${item.surname}`;
      } else {
        fullName = item.name;
      }
      return {
        value: fullName,
        label: fullName
      };
    });
    return compact(uniqBy(items, 'value'));
  }
  render() {
    return (
      <div className={classNames(styles.advancedInputWrapper)}>
        <Select
          value={this.state.choices}
          multi={true}
          placeholder={this.props.placeholder}
          onChange={this._handleChange}
          onFocus={this._handleFocus}
          onBlur={this._handleBlur}
          options={this._prepareItems()}
          className={classNames('uk-width-8-10', styles.advancedInput,
                      {[styles.colorGray]: !this.state.textInputFocused})} />
      </div>
    );
  }
}

AdvancedInput.propTypes = {
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
  onFocus: PropTypes.func,
  placeholder: PropTypes.string,
  items: PropTypes.array
};

AdvancedInput.defaultProps = {
  items: []
};


export default AdvancedInput;
