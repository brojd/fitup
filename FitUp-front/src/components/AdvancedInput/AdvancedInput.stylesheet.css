.advancedInputWrapper {
  margin-top: 21px; }

.advancedInput {
  margin-left: -1%; }

/*# sourceMappingURL=AdvancedInput.stylesheet.css.map */
