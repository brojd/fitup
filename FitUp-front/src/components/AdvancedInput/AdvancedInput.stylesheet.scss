@import "../../../sass_variables/colors";
@import "../../../sass_variables/widths";

.advancedInputWrapper {
  margin-top: 3 * $whitespace;
}
.advancedInput {
  margin-left: -1%;
}
