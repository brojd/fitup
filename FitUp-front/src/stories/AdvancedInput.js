import React from 'react';
import { storiesOf} from '@kadira/storybook';
import AdvancedInput from '../components/AdvancedInput/AdvancedInput.component';

storiesOf('AdvancedInput', module)
  .add('search by gyms', () => (
    <div className='uk-width-4-10'>
      <AdvancedInput placeholder='Search by gyms...'/>
    </div>
  ));

storiesOf('AdvancedInput', module)
  .add('search by trainers', () => (
    <div className='uk-width-4-10'>
      <AdvancedInput placeholder='Search by trainers...'/>
    </div>
  ));
