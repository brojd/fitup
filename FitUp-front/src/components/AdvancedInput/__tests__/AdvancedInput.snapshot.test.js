import React from 'react';
import AdvancedInput from '../AdvancedInput.component';
import renderer from 'react-test-renderer';

jest.mock('react-select');

test('AdvancedInput renders correctly excluding 3rd party react-select', () => {
  const component = renderer.create(
    <AdvancedInput />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
