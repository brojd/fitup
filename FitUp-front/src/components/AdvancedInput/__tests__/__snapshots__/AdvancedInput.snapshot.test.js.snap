exports[`test AdvancedInput renders correctly excluding 3rd party react-select 1`] = `
<div
  className="advancedInputWrapper" />
`;
